lib/global_uid/allocator.rb

```diff
@@ -0,0 +1,61 @@
+module GlobalUid
+  class Allocator
+    attr_reader :allocations, :max_size, :incrementing_by, :connection
+
+    def initialize(max_size: 5, incrementing_by:, connection:)
+      @allocations = []
+      @max_size = max_size
```

Comment:
Perhaps this should be called `@max_window_size` so it doesn't sound like you can, say, only allocate this many things ever

Reply:
Keep the comments coming! I'm appreciating the fresh set of eyes 🙏